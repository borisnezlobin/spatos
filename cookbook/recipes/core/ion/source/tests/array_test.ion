echo '# ANCHOR: standardArrays'
let array = []
echo @array
let array = [1]
for i in @array
    echo $i
end
let array = [1  [2 3]  4 [5 6]]
for i in @array
    echo $i
end
for i in [1  [2 3]  4 [5 6]]
    echo $i
end
echo '# ANCHOR_END: standardArrays'
echo '# ANCHOR: command_substitution'
let array_process = [ @(echo a  b c d    e) ]
for i in @array_process
    echo $i
end
for i in @(echo a  b c d    e)
    echo $i
end
echo '# ANCHOR_END: command_substitution'
echo '# ANCHOR: concatenation'
let new_array = [@array @array_process]
for i in @new_array
    echo $i
end
echo '# ANCHOR_END: concatenation'
echo '# ANCHOR: array_as_command_args'
echo @array
echo @array_process
echo @new_array
echo '# ANCHOR_END: array_as_command_args'
echo '# ANCHOR: slice_by_id'
let array = [ 1 2 3 ]
echo @array[0]
echo @array[1]
echo @array[2]
echo [ 1 2 3 ][0]
echo [ 1 2 3 ][1]
echo [ 1 2 3 ][2]
echo @(echo 1 2 3)[0]
echo @(echo 1 2 3)[1]
echo @(echo 1 2 3)[2]
echo '# ANCHOR_END: slice_by_id'
echo '# ANCHOR: slice_by_range'
let array = [ 1 2 3 4 5 ]
echo @array[0..1]
echo @array[0...1]
echo @array[..3]
echo @array[3..]
echo @array[..]
echo '# ANCHOR_END: slice_by_range'
echo '# ANCHOR: slice_with_vars'
for index in 0..3
  echo $(echo ðŸ˜‰ðŸ˜‰ðŸ˜‰)[$index]
end
echo '# ANCHOR_END: slice_with_vars'
echo '# ANCHOR: convert_to_string'
let array = [ 1 2 3 4 5 ]
let as_string = @array
echo $as_string
let array = [1 2 3 4 5 6 7 8 9 10]
echo @array[0 1 7..]
echo @array[2 3 5..]
echo @array[5.. 3 2]
echo @array[5..8 1..3 9 2]
echo '# ANCHOR_END: convert_to_string'
