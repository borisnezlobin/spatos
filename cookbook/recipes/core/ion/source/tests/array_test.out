# ANCHOR: standardArrays

1
1
2
3
4
5
6
1
2
3
4
5
6
# ANCHOR_END: standardArrays
# ANCHOR: command_substitution
a
b
c
d
e
a
b
c
d
e
# ANCHOR_END: command_substitution
# ANCHOR: concatenation
1
2
3
4
5
6
a
b
c
d
e
# ANCHOR_END: concatenation
# ANCHOR: array_as_command_args
1 2 3 4 5 6
a b c d e
1 2 3 4 5 6 a b c d e
# ANCHOR_END: array_as_command_args
# ANCHOR: slice_by_id
1
2
3
1
2
3
1
2
3
# ANCHOR_END: slice_by_id
# ANCHOR: slice_by_range
1
1 2
1 2 3
4 5
1 2 3 4 5
# ANCHOR_END: slice_by_range
# ANCHOR: slice_with_vars
ðŸ˜‰
ðŸ˜‰
ðŸ˜‰
# ANCHOR_END: slice_with_vars
# ANCHOR: convert_to_string
1 2 3 4 5
1 2 8 9 10
3 4 6 7 8 9 10
6 7 8 9 10 4 3
6 7 8 2 3 10 3
# ANCHOR_END: convert_to_string
